packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx

```diff
@@ -0,0 +1,293 @@
+          )}
+
+          <Controller
+            name="adminEmail"
+            control={newOrganizationFormMethods.control}
+            defaultValue=""
+            rules={{
+              required: t("must_enter_organization_admin_email"),
+            }}
+            render={({ field: { value } }) => (
+              <div className="flex">
+                <TextField
+                  containerClassName="w-full"
+                  placeholder="john@acme.com"
+                  name="adminEmail"
+                  label={t("admin_email")}
+                  defaultValue={value}
+                  onChange={(e) => {
+                    const domain = extractDomainFromEmail(e?.target.value);
+                    newOrganizationFormMethods.setValue("adminEmail", e?.target.value);
+                    newOrganizationFormMethods.setValue("adminUsername", e?.target.value.split("@")[0]);
+                    newOrganizationFormMethods.setValue("slug", domain);
+                    newOrganizationFormMethods.setValue(
+                      "name",
+                      domain.charAt(0).toUpperCase() + domain.slice(1)
+                    );
+                  }}
+                  autoComplete="off"
+                />
+              </div>
+            )}
+          />
+        </div>
+        <div className="mb-5">
+          <Controller
+            name="name"
+            control={newOrganizationFormMethods.control}
+            defaultValue=""
+            rules={{
+              required: t("must_enter_organization_name"),
+            }}
+            render={({ field: { value } }) => (
+              <>
+                <TextField
+                  className="mt-2"
+                  placeholder="Acme"
+                  name="name"
+                  label={t("organization_name")}
+                  defaultValue={value}
+                  onChange={(e) => {
+                    newOrganizationFormMethods.setValue("name", e?.target.value);
+                    if (newOrganizationFormMethods.formState.touchedFields["slug"] === undefined) {
+                      newOrganizationFormMethods.setValue("slug", slugify(e?.target.value));
+                    }
+                  }}
+                  autoComplete="off"
+                />
+              </>
+            )}
+          />
+        </div>
+
+        <div className="mb-5">
+          <Controller
+            name="slug"
+            control={newOrganizationFormMethods.control}
+            rules={{
+              required: "Must enter organization slug",
+            }}
+            render={({ field: { value } }) => (
+              <TextField
+                className="mt-2"
+                name="slug"
+                label={t("organization_url")}
+                placeholder="acme"
+                addOnSuffix={`.${process.env.NEXT_PUBLIC_WEBSITE_URL?.replace("https://", "")?.replace(
+                  "http://",
+                  ""
+                )}`}
```

Comment:
NIT: might be worth extracting this to a helper - seen this in a few places across this review